### PR TITLE
Prototype deployment

### PR DESCRIPTION
**Pull request by MartinLPetrauskas**
Changes necessary to smoothly install and have tinyHIPPO running for the midterm progress report

- Fixed typos
- Commented out circular imports and any exceptions that are raised in the code which cause the program to crash
- Added pertinent commands for tinyHIPPOstart and tinyHIPPOservice

Not fixed yet:
- Dashboard issues, mainly because they still use old 'capstone-ids' filepath

This PR will be a work in progress until all the issues are resolved

### CODE REVIEW

**Review by lucas-m98 — APPROVED** [review]
Looks good! I made some of the same path changes in my PR, but that shouldn't be an issue when merging.

File: setup/tinyHIPPOstart
Comment (lucas-m98) [inline setup/tinyHIPPOstart:14]: Where will is this going to be installed now? Also edit comment on line #4

Comment (MartinLPetrauskas) [reply setup/tinyHIPPOstart:14]: python3 and python3-pip are dependencies in the Makefile. So when the package is first installed on the router, opkg will check that the router has python3 and pip before installing tinyHIPPO.

So there is no need to make redundant calls here.

File: setup/tinyHIPPOstart
Comment (briweinstein) [inline setup/tinyHIPPOstart:21]: What would make this not-hacky?

File: src/dashboard/alerts/alert.py
Comment (briweinstein) [inline src/dashboard/alerts/alert.py:9]: Delete if not needed

File: src/dashboard/alerts/alert.py
Comment (briweinstein) [inline src/dashboard/alerts/alert.py:153]: Why is this commented out?

**Review by briweinstein — APPROVED** [review]

File: config.py
Comment (samir-elhelw76) [inline config.py:45]: Please remove this logic, I'll explain in our next meeting but I don't think it's necessary for us to check template values for the user in this portion of the code.

File: setup/tinyHIPPOstart
Comment (samir-elhelw76) [inline setup/tinyHIPPOstart:31]: Can this just be touch /etc/tinyHIPPO/alert_collection.json?

File: setup/tinyHIPPOstart
Comment (samir-elhelw76) [reply setup/tinyHIPPOstart:21]: Bumping what Bri said here, why is this hacky? Don't you need to copy these files to the necessary path? Or can we modify an environment variable or something that points OpenWrt to the directory where the source code for the luci dashboard is needed?

File: setup/tinyHIPPOstart
Comment (samir-elhelw76) [inline setup/tinyHIPPOstart:10]: General comment: Noticed that the base path /etc/tinyHIPPO is used a ton, we should pull this out to an environment variable and reference it in other parts of the code instead of repeating it.

File: src/dashboard/alerts/alert.py
Comment (samir-elhelw76) [reply src/dashboard/alerts/alert.py:9]: This is fixed in a separate PR

File: src/dashboard/alerts/alert.py
Comment (samir-elhelw76) [reply src/dashboard/alerts/alert.py:153]: Bumping Bri's comment, if the code doesn't work please try and fix and refrain from merging with master.

File: src/dashboard/luasrc/controller/ids-priv/ids_tab.lua
Comment (samir-elhelw76) [inline src/dashboard/luasrc/controller/ids-priv/ids_tab.lua:22]: Recognizing that you made edits to the luci code you should probably wait until Lucas' PR is merged, then pull and resubmit this PR since he does have a few edits in his branch that you replicate in yours.

File: config.py
Comment (samir-elhelw76) [inline config.py:11]: Does this create another config variable?

File: src/signature_detection/ip_signature.py
Comment (samir-elhelw76) [inline src/signature_detection/ip_signature.py:23]: I think we should raise custom exceptions here, and then in our entrypoint file we can catch those exceptions and log their messages in one place rather than having disparate logging calls throughout our code.

File: src/signature_detection/ip_signature.py
Comment (samir-elhelw76) [inline src/signature_detection/ip_signature.py:36]: See above comment

File: src/signature_detection/virustotal_checker.py
Comment (samir-elhelw76) [inline src/signature_detection/virustotal_checker.py:15]: See above comment

**Review by samir-elhelw76 — CHANGES_REQUESTED** [review]

File: src/dashboard/alerts/alert.py
Comment (lucas-m98) [reply src/dashboard/alerts/alert.py:153]: Once the circular import fix from #18 is added in it'll fix this, which hopefully will be merged right after this PR

File: setup/tinyHIPPOstart
Comment (MartinLPetrauskas) [reply setup/tinyHIPPOstart:21]: It is hacky because the Makefile and Openwrt Installation *should* take care of all the moving of files. So far, all my attempts to move the GUI related files have been unsuccessful. I should eventually figure it out, just more time is needed.

File: setup/tinyHIPPOstart
Comment (MartinLPetrauskas) [reply setup/tinyHIPPOstart:31]: Yes it can. Just fixed it. Will be pushing commit soon.

File: config.py
Comment (MartinLPetrauskas) [reply config.py:11]: This just instantiates a logging object which is saved in `self.log_event = logging.getLogger()` so that any class or file can just get the logging object and call it appropriately.

File: src/signature_detection/ip_signature.py
Comment (MartinLPetrauskas) [reply src/signature_detection/ip_signature.py:23]: Which entrypoint file? tiny_hippo_run.py? Raising exceptions is dangerous simply because if they are not caught, the program will crash/halt/stop.

File: config.py
Comment (samir-elhelw76) [reply config.py:11]: Okay, if it's a static value can it's declaration be moved to the constructor for the Config object?

File: src/signature_detection/ip_signature.py
Comment (samir-elhelw76) [reply src/signature_detection/ip_signature.py:23]: Yes the exceptions would be caught in tiny_hippo_run.py, then we could extract the messages that the exceptions return with a traceback and log it from there in one place though. This kind of logging just doesn't seem sustainable to me and should be abstracted if possible.

File: src/signature_detection/mac_address_signature.py
Comment (samir-elhelw76) [inline src/signature_detection/mac_address_signature.py:22]: See above comment.

**Review by samir-elhelw76 — CHANGES_REQUESTED** [review]

File: tiny_hippo_run.py
Comment (samir-elhelw76) [inline tiny_hippo_run.py:93]: The else block here does not work in the case that an exception is triggered on line 84 (triggered_rules will be empty). So it would make more sense if the logic was moved under the try portion of this block.

**Review by samir-elhelw76 — CHANGES_REQUESTED** [review]
Requested changes

**Review by samir-elhelw76 — APPROVED** [review]

**Pull request merged**
